### ISSUE TITLE
vue/static-class-names-order conflicts with Prettier

### ISSUE THREAD

**Issue by alex-lit** [vuejs/eslint-plugin-vue#1494]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.
  

**Tell us about your environment**

- **ESLint version:** 7.26.0
- **eslint-plugin-vue version:** 7.9.0
- **prettier version** 2.3.0
- **eslint-config-prettier version** 8.3.0
- **Node version:** 14.15.4
- **Operating System:** Deepin Linux

**Please show your full configuration:**

```js
module.exports = {
  extends: ["plugin:vue/recommended", "prettier"],

  plugins: ["vue", "prettier"],

  rules: {
    "vue/static-class-names-order": ["error"],
    "prettier/prettier": "error"
  }
};
```

**What did you do?**

```vue
<template>
  <div
    class="
        ui-container-sign-up__background-half
        ui-container-sign-up__background-half--sidebar
      "
  >
    <v-img
      class="ui-container-sign-up__image"
      eager
      height="100%"
      :src="image"
      width="100%"
    ></v-img>
  </div>

  <div
    class="
        ui-container-sign-up__background-half
        ui-container-sign-up__background-half--main
      "
  ></div>
</template>
```

**What did you expect to happen?**

There should be no errors

**What actually happened?**

![Screenshot_select-area_20210518145121](https://user-images.githubusercontent.com/5243967/118647049-94ffff00-b7e9-11eb-8654-fc076a944a69.jpg)

**Comment by ota-meshi** [comment]
If you want to use Prettier, you need to turn off some layout related rules for this plugin.

**Comment by developerdino** [comment]
We are having the same issue, but there doesn't seem to be any layout related rules to be able to turn of or modify for this plugin that I can see. Does anyone know what the settings would be to fix the above issue?

**Comment by neelance** [comment]
Here's a patch that makes the rule only check the order, not the whitespace:
```patch
diff --git a/node_modules/eslint-plugin-vue/lib/rules/static-class-names-order.js b/node_modules/eslint-plugin-vue/lib/rules/static-class-names-order.js
index bee820f..cea7ef7 100644
--- a/node_modules/eslint-plugin-vue/lib/rules/static-class-names-order.js
+++ b/node_modules/eslint-plugin-vue/lib/rules/static-class-names-order.js
@@ -33,26 +33,23 @@ module.exports = {
         if (!value) {
           return
         }
-        const classList = value.value
-        const classListWithWhitespace = classList.split(/(\s+)/)
+        const classList = value.value.split(/\s+/).filter((c) => c !== "");
 
-        // Detect and reuse any type of whitespace.
-        let divider = ''
-        if (classListWithWhitespace.length > 1) {
-          divider = classListWithWhitespace[1]
+        let sorted = true;
+        for (let i = 0; i < classList.length - 1; i++) {
+          if (classList[i] > classList[i + 1]) {
+            sorted = false;
+            break;
+          }
         }
 
-        const classListNoWhitespace = classListWithWhitespace.filter(
-          (className) => className.trim() !== ''
-        )
-        const classListSorted = classListNoWhitespace.sort().join(divider)
-
-        if (classList !== classListSorted) {
+        if (!sorted) {
+          classList.sort();
           context.report({
             node,
             loc: node.loc,
             message: 'Classes should be ordered alphabetically.',
-            fix: (fixer) => fixer.replaceText(value, `"${classListSorted}"`)
+            fix: (fixer) => fixer.replaceText(value, `"${classList.join(' ')}"`)
           })
         }
       }

```
